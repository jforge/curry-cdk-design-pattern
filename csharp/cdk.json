{
  "app": "dotnet run -p src/Dotnet/Dotnet.csproj"
}
