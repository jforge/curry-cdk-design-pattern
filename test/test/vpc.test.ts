import cdk = require('@aws-cdk/core');
import Vpc = require('../lib/vpc');
import '@aws-cdk/assert/jest';

test('VpcStack Tests', () => {
    const app = new cdk.App();
    // WHEN
    const stack = new Vpc.VpcStack(app, 'VpcStack');
    // THEN
    expect(stack).toHaveResource('AWS::EC2::VPC', {
      EnableDnsHostnames: true,
      EnableDnsSupport: true,
      InstanceTenancy: "default",
      Tags: [
        {
          "Key": "Name",
          "Value": "VpcStack/TheVPC"
        }
      ],
    });
});
