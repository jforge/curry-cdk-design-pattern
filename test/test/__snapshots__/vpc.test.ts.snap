// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests 1`] = `
Object {
  "Resources": Object {
    "TheVPC92636AB0": Object {
      "Properties": Object {
        "CidrBlock": "10.0.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
    "TheVPCIGWFA25CC08": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC",
          },
        ],
      },
      "Type": "AWS::EC2::InternetGateway",
    },
    "TheVPCPrivateSubnet1DefaultRouteD1B9E467": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "TheVPCPublicSubnet1NATGatewayC61D892B",
        },
        "RouteTableId": Object {
          "Ref": "TheVPCPrivateSubnet1RouteTableF6513BC2",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TheVPCPrivateSubnet1RouteTableAssociation46F1FFFC": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TheVPCPrivateSubnet1RouteTableF6513BC2",
        },
        "SubnetId": Object {
          "Ref": "TheVPCPrivateSubnet1Subnet571D3690",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TheVPCPrivateSubnet1RouteTableF6513BC2": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PrivateSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TheVPCPrivateSubnet1Subnet571D3690": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.128.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PrivateSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TheVPCPrivateSubnet2DefaultRoute52A1F245": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": Object {
          "Ref": "TheVPCPublicSubnet2NATGatewayB437CFAF",
        },
        "RouteTableId": Object {
          "Ref": "TheVPCPrivateSubnet2RouteTable9AC81FAC",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TheVPCPrivateSubnet2RouteTable9AC81FAC": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PrivateSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TheVPCPrivateSubnet2RouteTableAssociation336D47D1": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TheVPCPrivateSubnet2RouteTable9AC81FAC",
        },
        "SubnetId": Object {
          "Ref": "TheVPCPrivateSubnet2SubnetCC3D7013",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TheVPCPrivateSubnet2SubnetCC3D7013": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.192.0/18",
        "MapPublicIpOnLaunch": false,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PrivateSubnet2",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Private",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Private",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TheVPCPublicSubnet1DefaultRoute6D26543F": Object {
      "DependsOn": Array [
        "TheVPCVPCGWC9B93E30",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "TheVPCIGWFA25CC08",
        },
        "RouteTableId": Object {
          "Ref": "TheVPCPublicSubnet1RouteTable17DA183D",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TheVPCPublicSubnet1EIP4412F690": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "TheVPCPublicSubnet1NATGatewayC61D892B": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "TheVPCPublicSubnet1EIP4412F690",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "TheVPCPublicSubnet1Subnet770D4FF2",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet1",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "TheVPCPublicSubnet1RouteTable17DA183D": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet1",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TheVPCPublicSubnet1RouteTableAssociationE5186D77": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TheVPCPublicSubnet1RouteTable17DA183D",
        },
        "SubnetId": Object {
          "Ref": "TheVPCPublicSubnet1Subnet770D4FF2",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TheVPCPublicSubnet1Subnet770D4FF2": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            0,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.0.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet1",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TheVPCPublicSubnet2DefaultRouteFEB062B2": Object {
      "DependsOn": Array [
        "TheVPCVPCGWC9B93E30",
      ],
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "TheVPCIGWFA25CC08",
        },
        "RouteTableId": Object {
          "Ref": "TheVPCPublicSubnet2RouteTable3609F42C",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "TheVPCPublicSubnet2EIP5AAFF9FA": Object {
      "Properties": Object {
        "Domain": "vpc",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::EIP",
    },
    "TheVPCPublicSubnet2NATGatewayB437CFAF": Object {
      "Properties": Object {
        "AllocationId": Object {
          "Fn::GetAtt": Array [
            "TheVPCPublicSubnet2EIP5AAFF9FA",
            "AllocationId",
          ],
        },
        "SubnetId": Object {
          "Ref": "TheVPCPublicSubnet2Subnet73F96DA9",
        },
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet2",
          },
        ],
      },
      "Type": "AWS::EC2::NatGateway",
    },
    "TheVPCPublicSubnet2RouteTable3609F42C": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet2",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "TheVPCPublicSubnet2RouteTableAssociationB4B0A733": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "TheVPCPublicSubnet2RouteTable3609F42C",
        },
        "SubnetId": Object {
          "Ref": "TheVPCPublicSubnet2Subnet73F96DA9",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "TheVPCPublicSubnet2Subnet73F96DA9": Object {
      "Properties": Object {
        "AvailabilityZone": Object {
          "Fn::Select": Array [
            1,
            Object {
              "Fn::GetAZs": "",
            },
          ],
        },
        "CidrBlock": "10.0.64.0/18",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "VpcStack/TheVPC/PublicSubnet2",
          },
          Object {
            "Key": "aws-cdk:subnet-name",
            "Value": "Public",
          },
          Object {
            "Key": "aws-cdk:subnet-type",
            "Value": "Public",
          },
        ],
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "TheVPCVPCGWC9B93E30": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "TheVPCIGWFA25CC08",
        },
        "VpcId": Object {
          "Ref": "TheVPC92636AB0",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
  },
}
`;
